pyteamcity/future/artifact.py

```diff
@@ -0,0 +1,93 @@
+import fnmatch
+import os
+
+from .core.utils import parse_date_string
+
+
+class Artifact(object):
+    def __init__(self, build, path=''):
+        self.build = build
+        self.path = path
+        teamcity = self.build.build_query_set.teamcity
+        url = self.build.api_url + '/artifacts/metadata/' + self.path
+        res = teamcity.session.get(url)
+        res.raise_for_status()
+        self._data = res.json()
+
+    @property
+    def name(self):
+        return self._data['name']
+
+    def getsize(self):
+        return self._data.get('size')
+
+    @property
+    def size(self):
+        return self.getsize()
+
+    @property
+    def modification_time(self):
+        return parse_date_string(self._data['modificationTime'])
+
+    def splitext(self):
+        return os.path.splitext(self.name)
+
+    @property
+    def ext(self):
+        return self.splitext()[1]
+
+    def fnmatch(self, pattern):
+        return fnmatch.fnmatch(self.name, pattern)
+
+    @property
+    def content_href(self):
+        return self._data.get('content', {}).get('href')
+
+    def isdir(self):
+        return self.content_href is None
+
+    def isfile(self):
+        return self.content_href is not None
+
+    def __repr__(self):
+        return '<%s.%s: build.id=%r name=%r size=%r>' % (
+            self.__module__,
```

Comment:
Do you foresee having an `Artifact` object in multiple different modules? If not, I think the `__module__` would clutter the object repr.